src/wfinfo.c

```diff
@@ -1756,22 +1756,10 @@ NetLoad(VOID)
          //
          bNetShareLoad = TRUE;
 
-      } else {
-Fail:
-         //
-         // Disable the share buttons/menus
-         // Since WNetStat(NS_SHAREDLG) ret'd true, then we added the
-         // buttons.  Mistake.  Disable them now.
-         //
-         PostMessage(hwndToolbar, TB_ENABLEBUTTON, IDM_SHAREAS, FALSE);
          PostMessage(hwndToolbar, TB_ENABLEBUTTON, IDM_STOPSHARE, FALSE);

```

Comment:
Shouldn't this be removed? If I'm reading this right, the toolbar will never have an IDM_STOPSHARE item so this will always fail?

Reply:
I am not sure.
What I got from the code one can have the toolbar configured. This configuration is stored in the registry and is restored upon startup.

So it could happen, that one has configured toolbar with StopShare (even if it is useless), and it would be enabled if we omit this. The struct in tbar.c:55 is only an initial state

This is the same reason why we can not cleanup the bitmaps with toolbar icons.

Anyhow I might be mistaken, and that's why I kept it in. Hmm?

Reply:
Thanks for the explanation, makes sense.  I'd never looked at the toolbar customize case before.  Looking now though it seems to have some strange behaviors:

- If a toolbar button is already displayed (enabled or disabled), the user can configure its location
- Doing so will cause a disabled button to become enabled
- If the button is not displayed, it is only available to add if it would be enabled (see `HIDEIT`.)

The second point seems like a bug and the third seems very intentional but also strange.  It means that if somebody had customized their toolbar, they'll have stop sharing on it forever - but we can't add it for testing, because it's always disabled and hence can't be added.